panacea/query_client.go

```diff
@@ -306,6 +296,13 @@ func (q verifiedQueryClient) GetStoreData(ctx context.Context, storeKey string,
 	return result.Response.Value, nil
 }
 
+func (q verifiedQueryClient) getQueryBlockHeight(ctx context.Context) (int64, error) {
```

Comment:
Gets the height from the context, if the height exists in the context.
If notm get last block height from lightClient.

Reply:
It would be greater if we can do this without using context, since many practices are saying it's better to minimize the use of context. 
If we can pass a `targetHeight` parameter to the `GetStoreData()` function, it would be enough.

Reply:
OK. That would be good too.
If so, I will remove all contexts and modify it to specify the height only where it is needed.

I simply developed the function to be commonly applicable.

Reply:
I would like to be able to set the height selectively for each logic.

Reply:
However, I have a part that worries me.

The height to be used in the event and the height to be used in the client are currently implemented differently.
To speed up the client's response, the previous height of the last block was set.
To support both of these, the following changes must be made.
```go
type QueryClient interface {
	Close() error
	GetAccount(queryHeight int64, address string) (authtypes.AccountI, error)
	GetOracleRegistration(queryHeight int64,  uniqueID string, address string) (*oracletypes.OracleRegistration, error)
	GetLightBlock(height int64) (*tmtypes.LightBlock, error)
	GetCdc() *codec.ProtoCodec
	GetChainID() string
	GetOracleParamsPublicKey(queryHeight int64) (*btcec.PublicKey, error)
	GetDeal(queryHeight int64, dealID uint64) (*datadealtypes.Deal, error)
	GetCertificate(queryHeight int64, dealID uint64, dataHash string) (*datadealtypes.Certificate, error)
	GetLastBlockHeight() (int64, error)
}
```

Reply:
Hmm. If you think we need to change many source codes for using `queryHeight`, you can just use context as you did. I read this PR again, and it looks reasonable. Sorry for bothering you. So, the decision is up to you.

Reply:
I have been thinking about this part, and the conclusion will be modified so that the height is input as Youngjun initially suggested.
Using Middleware is convenient, but because it is hidden code, someone else brings confusion.

Reply:
Ah. I also forgot that `jwtAuthMiddleware` also uses verifiedQueryClient.
https://github.com/medibloc/panacea-oracle/pull/52/files#diff-6f125263042ee6bd63bdbbe9ea792e6d409132c4e91fc0e4553c668e9b9d99edR74

Then, it's more reasonable to use context to pass the query height to the `jwtAuthMiddleware`. Anyway, I'd like to say that the decision is up to you :)

Reply:
Ok, I'll decide to leave the code as is.

If I think this code is wrong, I'll fix it later. :)